### ISSUE TITLE
Maximization takes three steps for Chrome

### ISSUE THREAD

**Issue by hasantahir** [Hammerspoon/hammerspoon#3277]
As reported earlier in #2578, I notice that the Chrome app window takes three steps to maximize. For the rest of the apps, `hs.window:maximize()` works fine.

I have been using `hs.window.focusedWindow():maximize()` tied to the keyboard shortcut `⌘ + ⌥ + ⌃ + M` to maximise the windows.

Below is a screencast of what I observed:

https://user-images.githubusercontent.com/11236148/185097189-a055b605-1c04-46ab-97d7-30564bd0ee34.mp4

Cheers

**Comment by francoiscote** [comment]
may be related to #3224

**Comment by francoiscote** [comment]
see comments in the issues above. On my end this was caused by the Grammarly Desktop app. Quit it and restart Chrome.

**Comment by hasantahir** [comment]
Thanks for pointing to those comments. I tried to make sure all other apps were killed but got the same behaviour. Even in the incognito mode with all the extensions disabled, getting the same behaviour. Getting the same behaviour with Microsoft Edge. Firefox like other apps is working fine. 

Also running Karabiner elements but suspecting it has something to do with Chromium-based browsers.

**Comment by dasmurphy** [comment]
This may help... I use this vor maximising windows.

```lua
function maximizeWindow()
	-- full screen window
	local win = hs.window.focusedWindow()
	local f = win:frame()
	local screen = win:screen()
	local max = screen:frame()

	f.x = max.x
	f.y = max.y
	f.w = max.w
	f.h = max.h
	win:setFrame(f,0)
end
```

**Comment by hasantahir** [comment]
Thanks @dasmurphy 

Get the same issue with your routine. Everything except Chrome and Edge works fine.

**Comment by dasmurphy** [comment]
@hasantahir That is weird, since Chrome is working with this routine here fine.

**Comment by SethMilliken** [comment]
FWIW, I am experiencing these same symptoms (and not just for `hs.window:maximize()`, but also `hs.window:moveToUnit()` and `hs.window:setFrameInScreenBounds()`). For me it is with windows in these apps:

- 1Password
- Activity Monitor
- Finder
- Firefox
- Google Chrome
- Home
- iTerm
- MacVim
- Messages
- Music
- Safari
- System Preferences

Windows in these apps _do not_ exhibit the behavior (windows snap to correct size and position instantly):

- App Store
- Calendar
- Discord
- Firefox Nightly
- Hammerspoon
- Karabiner-Elements
- Mail
- Maps
- Microsoft Edge
- Notes
- Preview
- System Information
- Terminal
- Zoom

Pretty interesting that Firefox exhibits the problem, but not Firefox Nightly. And that Microsoft Edge is fine for me, but not for Hasan. I'm on macOS `12.6`.

Also,
- No Grammarly Desktop.
- Follow keyboard focus: Never

**Comment by Rhys-T** [comment]
Looks like this is probably an issue with the `AXEnhancedUserInterface` attribute. Full explanation and workaround posted here: https://github.com/Hammerspoon/hammerspoon/issues/3224#issuecomment-1294359070

**Comment by UseK** [comment]
Get the same issue with following environment.
- Intel Mac 2019
- macOS Sonoma 14.4.1
- Firefox 124.0.1

**Comment by sergei-dyshel** [comment]
Getting this on Sonoma with Firefox too.

**Comment by benjaminwelch** [comment]
I'm seeing this on Firefox (also Zen Browser based on Firefox) but doesn't appear to happen for any other apps I use.

I don't have Accessibility Keyboard enabled.

MacBookPro (M2 Pro)
macOS 13.4 (22F66)
Firefox: (137.0.2) newest

I'm using:
`hs.hotkey.bind({ "⌘", "⌥" }, "return", function() hs.window.focusedWindow():moveToUnit({ 0, 0, 1, 1 }) end)`

**Comment by Rhys-T** [comment]
Yeah, Firefox inflicts this on itself without any help from Accessibility Keyboard, VoiceOver, Grammarly, or anything else (https://github.com/Hammerspoon/hammerspoon/issues/3224#issuecomment-2155567686). Seems to be the result of a workaround for [a Voice Control-related issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1845364). There are links to a couple of relevant Bugzilla entries in <https://github.com/Hammerspoon/hammerspoon/issues/3224#issuecomment-2171827393>.
